{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['00000100']\n"
     ]
    }
   ],
   "source": [
    "#00\t0  00000000\n",
    "#AA\t1  00000001\n",
    "#CC\t2  00000010\n",
    "#GG\t3  00000011\n",
    "#TT\t4  00000100\n",
    "#ii 5  00000101\n",
    "\n",
    "#AC\t6  00000110\n",
    "#AG\t7  00000111\n",
    "#AT\t8  00001000\n",
    "#Ai\t9  00001001\n",
    "#CG\t10 00001010\n",
    "#CT\t11 00001011\n",
    "#Ci\t12 00001100\n",
    "#GT\t13 00001101\n",
    "#Gi\t14 00001110\n",
    "#Ti\t15 00001111\n",
    "\n",
    "gt_data = {}  # declare empty dictionary\n",
    "\n",
    "# allele_data is supposed to hold the columns REF and ALT from the VCF\n",
    "# allele_data[\"alt\"] is supposed to hold the list of alternative alleles. i.e. \"A,G,CCG,T etc\"\n",
    "allele_data = {}  # declare empty dictionary\n",
    "allele_data[\"alt\"] = {}\n",
    "allele_data[\"ref\"] = \"A\" # mockup vcf line with ref \"A\"\n",
    "\n",
    "\n",
    "allele_data[\"alt\"][1] = 'G' # mockup vcf line with first alt allele\n",
    "allele_data[\"alt\"][2] = 'T' # mockup vcf line with second alt allele\n",
    "\n",
    "# TODO Mockup VCF use all / for delimiter, but | is also possible\n",
    "#gt_data[\"GT\"] = \"0/1\"  # mockup  vcf line \n",
    "gt_data[\"GT\"] = \"2/2\"\n",
    "\n",
    "#mockup vcf line with indel Ci\n",
    "#allele_data['ref'] = 'GCTGCGT' #um zu testen ob Ai funktioniert\n",
    "                               \n",
    "#allele_data['alt'] = '' #alt is empty when it's a deletion \n",
    "#genotyp is 1/1 and alt is empty when its ii \n",
    "recoded_genotypes = [] # declare empty list\n",
    "if gt_data[\"GT\"] == \"0/0\": # homozygote, both reference allele\n",
    "    if allele_data[\"ref\"] == \"A\":\n",
    "        recoded_genotypes.append('00000001')  \n",
    "    elif allele_data[\"ref\"] == \"C\":\n",
    "        recoded_genotypes.append('00000010')\n",
    "    elif allele_data[\"ref\"] == \"G\":\n",
    "        recoded_genotypes.append('00000011')\n",
    "    elif allele_data[\"ref\"] == \"T\":\n",
    "        recoded_genotypes.append('00000100')\n",
    "        \n",
    "elif gt_data[\"GT\"][0] == gt_data[\"GT\"][2]: # homozygote, both identical alternative allele, get indices of alternative alleles and check them if they are equal\n",
    "    index_of_alt_allele = int(gt_data[\"GT\"][0])\n",
    "    if allele_data[\"alt\"][index_of_alt_allele] == \"A\":\n",
    "        recoded_genotypes.append('00000001')\n",
    "    elif allele_data[\"alt\"][index_of_alt_allele] == \"C\":\n",
    "        recoded_genotypes.append('00000010')\n",
    "    elif allele_data[\"alt\"][index_of_alt_allele] == \"G\":\n",
    "        recoded_genotypes.append('00000011')\n",
    "    elif  allele_data[\"alt\"][index_of_alt_allele] == \"T\":\n",
    "        recoded_genotypes.append('00000100')\n",
    "    elif allele_data[\"alt\"][index_of_alt_allele] == \"\": ##TODO: indels funktion und wenn es eine deletion in beide  chromosome ist\n",
    "        recoded_genotypes.append('00000101')\n",
    "\n",
    "        \n",
    "        \n",
    "elif gt_data[\"GT\"] == \"0/1\" or gt_data[\"GT\"] ==\"1/0\": # heterozygote\n",
    "    if allele_data[\"ref\"] == \"A\" and allele_data[\"alt\"] == \"C\": #ab 5 waren die kombination heterozygosit√§t und AC und CA sind gleich \n",
    "        recoded_genotypes.append('00000110')\n",
    "    elif allele_data[\"ref\"] == \"A\" and allele_data[\"alt\"] == \"G\":\n",
    "        recoded_genotypes.append('00000111')\n",
    "    elif allele_data[\"ref\"] == \"A\" and allele_data[\"alt\"] == \"T\":\n",
    "        recoded_genotypes.append('00001000')    \n",
    "    elif str(allele_data[\"ref\"]).startswith(\"A\") and allele_data[\"alt\"] == \"\": # If it starts with the letter \"A\", it's Ai\n",
    "        recoded_genotypes.append('00001001') \n",
    "        #TODO: indelin tabele auch  abschpeichern \n",
    "    elif allele_data[\"ref\"] == \"C\" and allele_data[\"alt\"] == \"G\":\n",
    "        recoded_genotypes.append('00001010')\n",
    "    elif allele_data[\"ref\"] == \"C\" and allele_data[\"alt\"] == \"T\":\n",
    "        recoded_genotypes.append('00001011')\n",
    "    elif str(allele_data[\"ref\"]).startswith(\"C\") and allele_data[\"alt\"] == \"\":  # If it starts with the letter \"C\", it's Ci\n",
    "        recoded_genotypes.append('00001100')\n",
    "        #TODO: indelin tabele auch  abschpeichern\n",
    "    elif allele_data[\"ref\"] == \"G\" and allele_data[\"alt\"] == \"T\":\n",
    "        recoded_genotypes.append('00001101')\n",
    "    elif str(allele_data[\"ref\"]).startswith(\"G\") and allele_data[\"alt\"] == \"\":  # If it starts with the letter \"G\", it's Gi\n",
    "        recoded_genotypes.append('00001110')\n",
    "        #TODO: indelin tabele auch  abschpeichern\n",
    "    elif str(allele_data[\"ref\"]).startswith(\"T\") and allele_data[\"alt\"] == \"\":  # If it starts with the letter \"T\", it's Ti\n",
    "        recoded_genotypes.append('00001111')\n",
    "        #TODO: indelin tabele auch  abschpeichern\n",
    "        \n",
    "else:\n",
    "    recoded_genotypes.append('00000000')\n",
    "print(recoded_genotypes)\n",
    "    \n",
    "#funktion um die Indels handeln "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
