{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['00001011']\n"
     ]
    }
   ],
   "source": [
    "#00\t0  00000000\n",
    "#AA\t1  00000001\n",
    "#CC\t2  00000010\n",
    "#GG\t3  00000011\n",
    "#TT\t4  00000100\n",
    "#xx 5  00000101\n",
    "\n",
    "#AC\t6  00000110\n",
    "#AG\t7  00000111\n",
    "#AT\t8  00001000\n",
    "#Ax\t9  00001001\n",
    "#CG\t10 00001010\n",
    "#CT\t11 00001011\n",
    "#Cx\t12 00001100\n",
    "#GT\t13 00001101\n",
    "#Gx\t14 00001110\n",
    "#Tx\t15 00001111\n",
    "\n",
    "#erstelle librrary\n",
    "\n",
    "\n",
    "bits = {\n",
    "    '00': '00000000',\n",
    "    'AA': '00000001',\n",
    "    'CC': '00000010',\n",
    "    'GG': '00000011',\n",
    "    'TT': '00000100',\n",
    "    'xx': '00000101',\n",
    "\n",
    "    'AC': '00000110',\n",
    "    'AG': '00000111',\n",
    "    'AT': '00001000',\n",
    "    'Ax': '00001001',\n",
    "    'CG': '00001010',\n",
    "    'CT': '00001011',\n",
    "    'Cx': '00001100',\n",
    "    'GT': '00001101',\n",
    "    'Gx': '00001110',\n",
    "    'Tx': '00001111',\n",
    "}\n",
    "\n",
    "gt_data = {}  # declare empty dictionary\n",
    "\n",
    "# allele_data is supposed to hold the columns REF and ALT from the VCF\n",
    "# allele_data[\"alt\"] is supposed to hold the list of alternative alleles. i.e. \"A,G,CCG,T etc\"\n",
    "allele_data = {}  # declare empty dictionary\n",
    "allele_data[\"alt\"] = {}\n",
    "allele_data[\"ref\"] = \"T\" # mockup vcf line with ref \"A\"\n",
    "\n",
    "\n",
    "allele_data[\"alt\"][1] = 'GC' # mockup vcf line with first alt allele\n",
    "allele_data[\"alt\"][2] = 'GCGGGGGTTTTT' # mockup vcf line with second alt allele, man kann noch mehrere kombination von allele hier \n",
    "                                        #weiter  hinzuf√ºgen\n",
    "allele_data[\"alt\"][3] = 'ACGT'\n",
    "allele_data[\"alt\"][4] = 'G'\n",
    "allele_data[\"alt\"][5] = 'GCG'  #usw\n",
    "\n",
    "# TODO Mockup VCF use all / for delimiter, but | is also possible\n",
    "#gt_data[\"GT\"] = \"0/1\"  # mockup  vcf line \n",
    "gt_data[\"GT\"] = \"4/5\"\n",
    "\n",
    "#mockup vcf line with indel Ci\n",
    "#allele_data['ref'] = 'GCTGCGT' #um zu testen ob Ai funktioniert\n",
    "                               \n",
    "#allele_data['alt'] = '' #alt is empty when it's a deletion\n",
    "\n",
    "\n",
    "recoded_genotypes = [] # declare empty list\n",
    "alleles= []\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "gt_allele_1 = int(gt_data[\"GT\"][0]) # i.e. 2<--/1\n",
    "gt_allele_2 = int(gt_data[\"GT\"][2]) # i.e. 2/-->1\n",
    "\n",
    "alleles.append(allele_data[\"alt\"][1])\n",
    "\n",
    "\n",
    "recoded_genotypes.append(bits['CT'])\n",
    "print(recoded_genotypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
