{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chrom: 20, Pos: 14370, Sample: NA00001, Genotype: AA, Recoded genotype: 00000001\n",
      "Chrom: 20, Pos: 14370, Sample: NA00002, Genotype: AA, Recoded genotype: 00000001\n",
      "Chrom: 20, Pos: 14370, Sample: NA00003, Genotype: AA, Recoded genotype: 00000001\n",
      "['00000001', '00000001', '00000001']\n",
      "This is homozygous\n",
      "Sample (person) 0 and sample (person) 0 share the same homozygous region\n",
      "Sample (person) 0 and sample (person) 1 share the same homozygous region\n",
      "Sample (person) 0 and sample (person) 2 share the same homozygous region\n",
      "This is homozygous\n",
      "Sample (person) 1 and sample (person) 0 share the same homozygous region\n",
      "Sample (person) 1 and sample (person) 1 share the same homozygous region\n",
      "Sample (person) 1 and sample (person) 2 share the same homozygous region\n",
      "This is homozygous\n",
      "Sample (person) 2 and sample (person) 0 share the same homozygous region\n",
      "Sample (person) 2 and sample (person) 1 share the same homozygous region\n",
      "Sample (person) 2 and sample (person) 2 share the same homozygous region\n"
     ]
    }
   ],
   "source": [
    "bits = {\n",
    "    '00': '00000000',\n",
    "    'AA': '00000001',\n",
    "    'CC': '00000010',\n",
    "    'GG': '00000011',\n",
    "    'TT': '00000100',\n",
    "    'xx': '00000101',\n",
    "\n",
    "    'AC': '00000110',\n",
    "    'AG': '00000111',\n",
    "    'AT': '00001000',\n",
    "    'Ax': '00001001',\n",
    "    'CG': '00001010',\n",
    "    'CT': '00001011',\n",
    "    'Cx': '00001100',\n",
    "    'GT': '00001101',\n",
    "    'Gx': '00001110',\n",
    "    'Tx': '00001111',\n",
    "}\n",
    "\n",
    "filename = 'wikipedia-example.vcf'\n",
    "# filename = 'short.vcf'\n",
    "\n",
    "with open(filename) as file: #lese vcf datei\n",
    "    line = file.readline()\n",
    "    while line:\n",
    "        if line.startswith('#CHROM'):\n",
    "            # #CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tNA12891\tNA12892\tNA12878\n",
    "            samples = line.split(sep='\\t')[9:] #skip first 9 entries, intesrrissieren uns nur wie viele Samples da vorhanden sind, wir wollen genotypes von den samples wissen\n",
    "        elif not line.startswith('#'):\n",
    "#             print(line.strip())\n",
    "#1\t742429\trs3094315\tG\tA,T\t.\tPASS\tAA=g;DP=132;HM2\tGT:GQ:DP\t1|1:100:38\t0|0:59:30\t0|1:100:44\n",
    "\n",
    "            chrom, position, ref, alt, formatstring = list(line.split(sep='\\t')[i] for i in [0, 1, 3, 4, 8])\n",
    "            genotypes = line.split(sep='\\t')[9:] #skip all before the 9th\n",
    "\n",
    "            datas = formatstring.split(':')\n",
    "            alleles = [ref] + alt.split(',') #concatieniere die ref liste, und alt liste \n",
    "            gt_data = {}\n",
    "            \n",
    "            recoded_genotypes = []\n",
    "            for sample in samples: # wir gehen durch die maximale anzahl von samples, welche definiert wird in #CHROM zeil am anfang der VCF datei\n",
    "                for genotype in genotypes: #\n",
    "                    if genotype: # This means checking if the string genotype is not empty\n",
    "                        for idx, data in enumerate(datas):\n",
    "                            gt_data[data] = genotype.split(':')[idx]\n",
    "                gt_allele_1 = int(gt_data[\"GT\"][0]) # i.e. 2<--/1\n",
    "                gt_allele_2 = int(gt_data[\"GT\"][2]) # i.e. 2/-->1\n",
    "                \n",
    "\n",
    "\n",
    "                real_alleles= [] #list with explicit allele to be sorted\n",
    "\n",
    "                real_alleles.append(alleles[gt_allele_1])\n",
    "                real_alleles.append(alleles[gt_allele_2])\n",
    "\n",
    "\n",
    "\n",
    "                for idx, real_allele in enumerate(real_alleles):\n",
    "                    if len(real_alleles[idx]) != 1: #gucken genotypen an , bezieht siech auf allele\n",
    "                        real_alleles[idx] = 'x'\n",
    "\n",
    "                real_alleles.sort()\n",
    "                allele_string = real_alleles[0] + real_alleles[1]\n",
    "                recoded_genotype = bits[allele_string]\n",
    "                recoded_genotypes.append(recoded_genotype)\n",
    "                \n",
    "                \n",
    "                print('Chrom: %s, Pos: %s, Sample: %s, Genotype: %s, Recoded genotype: %s' % (chrom, position, sample.rstrip(), allele_string, recoded_genotype))\n",
    "            print(recoded_genotypes)\n",
    "            for idx, recoded_genotype_2 in enumerate(recoded_genotypes):\n",
    "                if int(recoded_genotype_2, 2) <= 5:\n",
    "                    print('This is homozygous')\n",
    "                    for idx_2, recoded_genotype_3 in enumerate(recoded_genotypes):\n",
    "                        if recoded_genotype_2 == recoded_genotype_3:\n",
    "                            print('Sample (person) %i and sample (person) %i share the same homozygous region' % (idx, idx_2))\n",
    "            \n",
    "        line = file.readline()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
